#!/usr/bin/env python
PACKAGE = "zvision_lidar_pointcloud"

from math import pi
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("time_offset", double_t,  0, "A manually calibrated offset (in seconds) to add to the timestamp before publication of a message.", 0.0, -1.0, 1.0)
gen.add("x_trans", double_t,  1, "A manually calibrated offset (in meter) to x trans.", .0, -1.0, 1.0)
gen.add("y_trans", double_t,  2, "A manually calibrated offset (in meter) to y trans.", .0, -1.0, 1.0)
gen.add("z_trans", double_t,  3, "A manually calibrated offset (in meter) to z trans.", .0, -1.0, 1.0)

gen.add("x_rotation", double_t,  4, "A manually calibrated offset (in degree) to x rotation.", 0.0, -120.0, 120.0)
gen.add("y_rotation", double_t,  5, "A manually calibrated offset (in degree) to y rotation.", 0.0, -120.0, 120.0)
gen.add("z_rotation", double_t,  6, "A manually calibrated offset (in degree) to z rotation.", 0.0, -120.0, 120.0)

exit(gen.generate(PACKAGE, "cloud_node", "CloudNode"))
